### PR TITLE
Add gdal shared build

### PR DESCRIPTION
**Pull request by saiarcot895**
This adds support for making a shared build of gdal, hdf4, and netcdf on i686-w64-mingw32 (currently). I've confirmed that this builds on i686-w64-mingw32, and will be adding support for x86_64-w64-mingw32 shortly.

### CODE REVIEW

File: src/gdal.mk [hunk 1]
```diff
@@ -102,5 +98,3 @@ endef
 
 # Can't use $(PKG)_BUILD_SHARED here as $(PKG)_BUILD_i686-w64-mingw32 has a
 # higher precedence.
```
Comment (TimothyGu) [inline src/gdal.mk]: You should now remove this comment.

File: src/hdf4.mk [hunk 2]
```diff
@@ -21,19 +21,25 @@ define $(PKG)_BUILD
     cd '$(1)' && $(LIBTOOLIZE) --force
     cd '$(1)' && autoreconf --install
     cd '$(1)' && ./configure \
-        --host='$(TARGET)' \
-        --build="`config.guess`" \
-        --disable-shared \
+    $(MXE_CONFIGURE_OPTS) \
```
Comment (TimothyGu) [inline src/hdf4.mk]: Indentation

File: src/hdf4.mk [hunk 3]
```diff
@@ -21,19 +21,25 @@ define $(PKG)_BUILD
...
         --disable-fortran \
         --disable-netcdf \
-        --prefix='$(PREFIX)/$(TARGET)' \
-        CPPFLAGS="-DH4_F77_FUNC\(name,NAME\)=NAME -DH4_BUILT_AS_STATIC_LIB=1"
-    $(MAKE) -C '$(1)'/hdf/src -j '$(JOBS)'
+    $(if $(BUILD_STATIC), \
```
Comment (TimothyGu) [inline src/hdf4.mk]: ditto

File: src/netcdf.mk [hunk 4]
```diff
@@ -26,20 +26,14 @@ define $(PKG)_BUILD
...
-endef
 
-$(PKG)_BUILD_x86_64-w64-mingw32 =
-$(PKG)_BUILD_i686-w64-mingw32 = $(subst --enable-hdf4, --disable-hdf4,\
-                                $(subst -lmfhdf -ldf,,$($(PKG)_BUILD)))
+    $(MAKE) -C '$(1)' -j '$(JOBS)' \
+    LDFLAGS=-no-undefined
 
-$(PKG)_BUILD_SHARED =
+    $(MAKE) -C '$(1)' -j 1 install
```
Comment (TimothyGu) [inline src/netcdf.mk]: Ditto

**Comment by saiarcot895** [comment]
Done. I've also combined the two conditionals in `hdf4.mk`.

**Comment by hamzaalloush** [comment]
Sorry my latest gdal commit had commented out OGR, please remove comment as that from #OGR to OGR so symbols get exported

**Comment by saiarcot895** [comment]
OGR has been uncommented, and merge conflicts have been resolved. It looks like this is now only adding gdal shared build.

**Comment by TimothyGu** [comment]
Thanks!

**Pull request merged**